core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -200,32 +202,47 @@ class TaskMetrics private[spark] () extends Serializable {
 
 
   import InternalAccumulator._
-  @transient private[spark] lazy val nameToAccums = LinkedHashMap(
```

Comment:
It looks like the use of `LinkedHashMap` was added by @cloud-fan in #12612. As far as I can tell we don't actually rely on the ordering of the entries in this map, so I didn't preserved the use of `LinkedHashMap`.